magma/magma
# Update Orc8r’s subscriberdb servicer to support flat subscriber digests

Signed-off-by: Yuanyuting Wang <yw3241@columbia.edu>

## Summary


1. Modify protos for `subscriberdb` servicer to include digest/no-update short-circuiting logic-related fields in `ListSubscribersRequest` and `ListSubscribersResponse`
2. Modify `ListSubscribers` rpc implementation to include logic for comparing gateway and cloud digests to decide whether to send over updated subscriber data
3. Add test case for communicating with the servicer using digests

## Test Plan


1. Run all existing tests. All should pass
2. Add new test case in `subscriberdb_servicer_test.go` that tests the `subscriberdb` cloud servicer with digests. This test should also pass

## Additional Information
